### DISCUSSION TITLE
Missing Data from Cocktails API Request

### DISCUSSION THREAD

**Discussion by raddacle** [karlomikus/bar-assistant#145] [Q&A] [answered]
When I pull a /cocktails Get request, none of the cocktails include their glass type, even though they're identified in the database. The schema shows that it should be apart of it, but I can't find how. Any ideas? @karlomikus @augustf 

Here's the result of the request: /bar/api/cocktails?filter%5Btag_id%5D=4&per_page=2&page=1

Result:

> {
>     "data": [{
>         "id": 102,
>         "name": "20th Century",
>         "slug": "20th-century",
>         "instructions": "1. Combine all ingredients with ice and shake\n2. Strain into a coupe, serve up",
>         "garnish": "Lemon twist",
>         "description": "The 20th Century is a cocktail created in 1937 by a British bartender named C.A. Tuck, and named in honor of the celebrated 20th Century Limited train which ran between New York City and Chicago from 1902 until 1967.",
>         "source": "https:\/\/en.wikipedia.org\/wiki\/20th_Century_(cocktail)",
>         "has_public_link": false,
>         "public_id": null,
>         "main_image_id": 226,
>         "images": [{
>             "id": 226,
>             "file_path": "cocktails\/20th-century_oHkdoE.jpg",
>             "url": "https:\/\/bar.[webaddress.com]\/bar\/uploads\/cocktails\/20th-century_oHkdoE.jpg",
>             "copyright": "Imbibe Magazine",
>             "sort": 0,
>             "placeholder_hash": "DykODwIIt3iHdnh0i0aniId3dwKIxnYP"
>         }],
>         "tags": ["Gin"],
>         "user_id": 1,
>         "user_name": null,
>         "user_rating": null,
>         "average_rating": 0,
>         "short_ingredients": ["Gin", "White Cr\u00e8me de Cacao", "Lillet Blanc", "Lemon juice"],
>         "ingredients": [{
>             "sort": 1,
>             "amount": 45,
>             "units": "ml",
>             "optional": false,
>             "ingredient_id": 76,
>             "name": "Gin",
>             "ingredient_slug": "gin"
>         }, {
>             "sort": 2,
>             "amount": 22.5,
>             "units": "ml",
>             "optional": false,
>             "ingredient_id": 31,
>             "name": "White Cr\u00e8me de Cacao",
>             "ingredient_slug": "white-creme-de-cacao"
>         }, {
>             "sort": 3,
>             "amount": 22.5,
>             "units": "ml",
>             "optional": false,
>             "ingredient_id": 125,
>             "name": "Lillet Blanc",
>             "ingredient_slug": "lillet-blanc"
>         }, {
>             "sort": 4,
>             "amount": 15,
>             "units": "ml",
>             "optional": false,
>             "ingredient_id": 59,
>             "name": "Lemon juice",
>             "ingredient_slug": "lemon-juice"
>         }],
>         "main_ingredient_name": "Gin",
>         "created_at": "2023-07-22 10:47:50",
>         "method": {
>             "id": 1,
>             "name": "Shake",
>             "dilution_percentage": 25
>         },
>         "abv": 20.91
>     }, {
>         "id": 103,
>         "name": "Alaska",
>         "slug": "alaska",
>         "instructions": "1. Combine all ingredients with ice in a mixing glass and stir at length, until the sides of the glass are frosty\n2. Strain into a cocktail glass and serve up",
>         "garnish": "",
>         "description": "One of the great Chartreuse cocktails and a fundamental three-ingredient recipe",
>         "source": "https:\/\/tuxedono2.com\/alaska-cocktail-recipe",
>         "has_public_link": false,
>         "public_id": null,
>         "main_image_id": 227,
>         "images": [{
>             "id": 227,
>             "file_path": "cocktails\/alaska_iWcNcb.jpg",
>             "url": "https:\/\/bar.[webaddress.com]\/bar\/uploads\/cocktails\/alaska_iWcNcb.jpg",
>             "copyright": "Punch",
>             "sort": 0,
>             "placeholder_hash": "2DgaFwZnh4eAh3iJiGeXZ4eJd2QIh5YA"
>         }],
>         "tags": ["Gin"],
>         "user_id": 1,
>         "user_name": null,
>         "user_rating": null,
>         "average_rating": 0,
>         "short_ingredients": ["Gin", "Yellow Chartreuse", "Orange bitters"],
>         "ingredients": [{
>             "sort": 1,
>             "amount": 45,
>             "units": "ml",
>             "optional": false,
>             "ingredient_id": 76,
>             "name": "Gin",
>             "ingredient_slug": "gin"
>         }, {
>             "sort": 2,
>             "amount": 15,
>             "units": "ml",
>             "optional": false,
>             "ingredient_id": 43,
>             "name": "Yellow Chartreuse",
>             "ingredient_slug": "yellow-chartreuse"
>         }, {
>             "sort": 3,
>             "amount": 1,
>             "units": "dash",
>             "optional": false,
>             "ingredient_id": 21,
>             "name": "Orange bitters",
>             "ingredient_slug": "orange-bitters"
>         }],
>         "main_ingredient_name": "Gin",
>         "created_at": "2023-07-22 10:47:50",
>         "method": {
>             "id": 2,
>             "name": "Stir",
>             "dilution_percentage": 20
>         },
>         "abv": 33.23
>     }],
>     "links": {
>         "first": "http:\/\/bar-assistant:3000\/api\/cocktails?page=1",
>         "last": "http:\/\/bar-assistant:3000\/api\/cocktails?page=16",
>         "prev": null,
>         "next": "http:\/\/bar-assistant:3000\/api\/cocktails?page=2"
>     },
>     "meta": {
>         "current_page": 1,
>         "from": 1,
>         "last_page": 16,
>         "links": [{
>             "url": null,
>             "label": "&laquo; Previous",
>             "active": false
>         }, {
>             "url": "http:\/\/bar-assistant:3000\/api\/cocktails?page=1",
>             "label": "1",
>             "active": true
>         }, {
>             "url": "http:\/\/bar-assistant:3000\/api\/cocktails?page=2",
>             "label": "2",
>             "active": false
>         }, {
>             "url": "http:\/\/bar-assistant:3000\/api\/cocktails?page=3",
>             "label": "3",
>             "active": false
>         }, {
>             "url": "http:\/\/bar-assistant:3000\/api\/cocktails?page=4",
>             "label": "4",
>             "active": false
>         }, {
>             "url": "http:\/\/bar-assistant:3000\/api\/cocktails?page=5",
>             "label": "5",
>             "active": false
>         }, {
>             "url": "http:\/\/bar-assistant:3000\/api\/cocktails?page=6",
>             "label": "6",
>             "active": false
>         }, {
>             "url": "http:\/\/bar-assistant:3000\/api\/cocktails?page=7",
>             "label": "7",
>             "active": false
>         }, {
>             "url": "http:\/\/bar-assistant:3000\/api\/cocktails?page=8",
>             "label": "8",
>             "active": false
>         }, {
>             "url": "http:\/\/bar-assistant:3000\/api\/cocktails?page=9",
>             "label": "9",
>             "active": false
>         }, {
>             "url": "http:\/\/bar-assistant:3000\/api\/cocktails?page=10",
>             "label": "10",
>             "active": false
>         }, {
>             "url": null,
>             "label": "...",
>             "active": false
>         }, {
>             "url": "http:\/\/bar-assistant:3000\/api\/cocktails?page=15",
>             "label": "15",
>             "active": false
>         }, {
>             "url": "http:\/\/bar-assistant:3000\/api\/cocktails?page=16",
>             "label": "16",
>             "active": false
>         }, {
>             "url": "http:\/\/bar-assistant:3000\/api\/cocktails?page=2",
>             "label": "Next &raquo;",
>             "active": false
>         }],
>         "path": "http:\/\/bar-assistant:3000\/api\/cocktails",
>         "per_page": 2,
>         "to": 2,
>         "total": 31
>     }
> }

**Comment by karlomikus** [comment]
As a optimization step, glass info is currently only shown on cocktail details, ie cocktails/id

**Comment by raddacle** [comment]
Oh, so I'll need to make a call for every cocktail. I'm working on a web page that creates a sleek menu, organized by tag/collection from the api, that's meant to be given out to guests as a link or qr code for easy access to the menu., and is meant to be used in conjunction with Salt Rim. As each menu item will have an icon of the glass type, it seems like it'll have to send a batch of api calls for every drink on the menu regularly.

**Reply by karlomikus** [reply, marked as answer]
Ill add on demand relationship loading via query strings in next version, should solve this problem and its an easy fix.

**Reply by raddacle** [reply]
That would be fantastic, thank you!
